seed/seed20.go

```diff
@@ -297,26 +302,51 @@ func (s *seed20) copySnapAndComponents(sn *Snap, destSeedDir string, opts CopyOp
 	if snapAsserted {
 		decl, ok := s.snapDeclsByID[sn.ID()]
 		if !ok {
-			return nil, fmt.Errorf("internal error: missing snap-declaration for asserted snap: %s", sn.SnapName())
+			return nil, nil, fmt.Errorf("internal error: missing snap-declaration for asserted snap: %s", sn.SnapName())
 		}
 		assertions = append(assertions, decl)
 
 		rev, ok := s.snapRevsByID[sn.ID()]
 		if !ok {
-			return nil, fmt.Errorf("internal error: missing snap-revision for asserted snap: %s", sn.SnapName())
+			return nil, nil, fmt.Errorf("internal error: missing snap-revision for asserted snap: %s", sn.SnapName())
 		}
 		assertions = append(assertions, rev)
 	}
 
-	inModel := snapInModel(sn, s.modelSnaps)
+	snapInModel := snapInModel(sn, s.modelSnaps)
 
-	snapDest := destination(snapAsserted, inModel, filepath.Base(sn.Path))
+	snapDest := destination(snapAsserted, snapInModel, filepath.Base(sn.Path))
 	if err := osutil.CopyFile(sn.Path, snapDest, osutil.CopyFlagOverwrite); err != nil {
-		return nil, fmt.Errorf("cannot copy snap: %w", err)
+		return nil, nil, fmt.Errorf("cannot copy snap: %w", err)
+	}
+
+	var optSnap *internal.Snap20
+	for _, os := range s.optSnaps {
+		if os.Name == sn.SnapName() {
+			cp := *os
+			// make a copy so we don't mess with the original options snap
+			optSnap = &cp
+			optSnap.Components = append([]internal.Component20(nil), cp.Components...)
+		}
+	}
+
+	removeOptionsComponent := func(name string) {
+		if optSnap == nil {
+			return
+		}
+
+		comps := make([]internal.Component20, 0, len(optSnap.Components))
+		for _, comp := range optSnap.Components {
+			if comp.Name != name {
+				comps = append(comps, comp)
+			}
+		}
+		optSnap.Components = comps
```

Comment:
can't we append the components that we copy instead of removing the ones that we don't ?

Reply:
Yes, that approach works too. I'll change it.